packages/@aws-cdk/toolkit-lib/lib/api/io/private/codes.ts

```diff
@@ -145,24 +242,45 @@ export const CODES = {
     description: 'Provides bootstrap times',
     interface: 'Duration',
   }),
-  CDK_TOOLKIT_I9900: make.info({
+  CDK_TOOLKIT_I9100: make.info<BootstrapEnvironmentProgress>({
+    code: 'CDK_TOOLKIT_I9100',
+    description: 'Bootstrap progress',
+    interface: 'BootstrapEnvironmentProgress',
+  }),
+
+  CDK_TOOLKIT_I9900: make.result<{ environment: cxapi.Environment }>({
     code: 'CDK_TOOLKIT_I9900',
     description: 'Bootstrap results on success',
+    interface: 'cxapi.Environment',
   }),
-  CDK_TOOLKIT_E9900: make.error({
+  CDK_TOOLKIT_E9900: make.error<ErrorPayload>({
     code: 'CDK_TOOLKIT_E9900',
     description: 'Bootstrap failed',
+    interface: 'ErrorPayload',
   }),
 
   // Assembly codes
+  CDK_ASSEMBLY_I0010: make.debug<never>({
```

Comment:
for my information, why `never` added here?

Reply:
Not needed, but it's the same as leaving it out.